lib/galaxy/model/migrate/versions/0155_add_content_update_time.py

```diff
@@ -0,0 +1,105 @@
+"""
+Adds trigger to dataset table to update history.update_time when
+contents are changed.
+"""
+from __future__ import print_function
+
+import logging
+
+from sqlalchemy import DDL, MetaData
+
+log = logging.getLogger(__name__)
+metadata = MetaData()
+
+
+def upgrade(migrate_engine):
+    print(__doc__)
+    metadata.bind = migrate_engine
+    metadata.reflect()
+    drop_update_trigger(migrate_engine)
+    install_update_trigger(migrate_engine)
+
+
+def downgrade(migrate_engine):
+    metadata.bind = migrate_engine
+    metadata.reflect()
+    drop_update_trigger(migrate_engine)
+
+
+def not_pg(self, target, bind, state, **kw):
+    return bind.engine.name != 'postgresql'
+
+
+def install_update_trigger(migrate_engine):
+    """Installs trigger on database table to update history table
+    when contents have changed. Installs a function and a trigger
+    for postgres, other sql variants only require the trigger def
+    """
+
+    pg_create_trigger = DDL("""
+        CREATE FUNCTION update_history_content_update_time()
+            RETURNS trigger
+            LANGUAGE 'plpgsql'
+        AS $BODY$
+        begin
```

Comment:
Can you use uppercase for SQL keywords here and below?

Reply:
Yep.